crates/vm/levm/src/precompiles.rs

```diff
@@ -501,12 +501,186 @@ pub fn ecmul(
     }
 }
 
-fn ecpairing(
-    _calldata: &Bytes,
-    _gas_for_call: u64,
-    _consumed_gas: &mut u64,
+pub fn ecpairing(
+    calldata: &Bytes,
+    gas_for_call: u64,
+    consumed_gas: &mut u64,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    // The input must always be a multiple of 192 (6 32-byte values)
+    if calldata.len() % 192 != 0 {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    let groups_number = calldata.len() / 192;
+
+    // Consume gas
+    let gas_cost = gas_cost::ecpairing(groups_number)?;
+    increase_precompile_consumed_gas(gas_for_call, gas_cost, consumed_gas)?;
+
+    let mut mul: FieldElement<Degree12ExtensionField> = QuadraticExtensionFieldElement::one();
+    for group_number in 0..groups_number {
```

Comment:
And this would be like input index, right?

Reply:
Yes, done in [Replace group terminology, use input instead](https://github.com/lambdaclass/ethrex/pull/1560/commits/7273981de05aa4e87ad180498ce4deb8dfaae7ed)